kernelci/build.py

```diff
@@ -1211,6 +1230,43 @@ def install(self, verbose=False):
         return super().install(verbose)
 
 
+class FetchFirmware(Step):
+
+    @property
+    def name(self):
+        return 'firmware'
+
+    def run(self, jopt=None, verbose=False, opts=None):
+        """Fetch linux-firmware repository
+
+        If kernel have CONFIG_EXTRA_FIRMWARE enabled we need to fetch
+        fresh snapshot of linux-firmware git repository, otherwise
+        kernel will not be able to build
+        """
+        # CONFIG_EXTRA_FIRMWARE_DIR need absolute path
+        full_path = os.path.abspath(self._output_path)
+        fwdir = os.path.join(full_path, 'linux-firmware')
+        key = self._kernel_config_getkey('CONFIG_EXTRA_FIRMWARE')
+        if key and len(key) < 3:
+            print("External firmware not required")
+            return self._add_run_step(True)
+        print("Fetching firmware")
+        repourl = 'git://git.kernel.org/pub/scm/linux/kernel/git/\
+firmware/linux-firmware.git'
+        tree = kernelci.config.build.Tree('firmware', repourl)
+        config = kernelci.config.build.BuildConfig('firmware', tree,
+                                                   'main', dict())
+        update_repo(config, fwdir, shallow=True)
```

Comment:
Well, no that's really a hack.  The `update_repo()` function is expecting a build configuration for a kernel tree.  Here it's being stretched to use another Git repo (firmwares), so I think that's going to be a source of bugs.